net/devif/devif_callback.c

```diff
@@ -43,8 +43,8 @@
  * Private Data
  ****************************************************************************/
 
-#ifndef CONFIG_NET_ALLOC_CONNS
-static struct devif_callback_s g_cbprealloc[CONFIG_NET_NACTIVESOCKETS];
+#if CONFIG_NET_PREALLOC_SOCKETS > 0
```

Comment:
should we change NET_PREALLOC_SOCKETS  to NET_PREALLOC_CALLBACK or NET_PREALLOC_DEVIF_CALLBACK?

Reply:
I could rename this, but it is currently defined in the "Socket Support" menu in Kconfig.  
Maybe it should be moved too? But I can't decide where it would be best.

Reply:
Move to devif/Kconfig or net/kconfig?

Reply:
Just renamed for the moment. I didn't like any other place to move to.
(devif/Kconfig does not exist and devif/Kconfig would be a bit chaotic...)